### ISSUE TITLE
progress?

### ISSUE THREAD

**Issue by twhitbeck** [angular/router#317]
hi @btford 
I'd like to request, in the most non demanding way, an update on the project. I see some work on the angular2 repo, and the most recent meeting notes mention that the new router will not be ready for angular 1.4. That's sad. Could you shout out a little update? Is angular2 a priority over 1.x now? It seemed like it was the other way around at one point.

Thanks!

**Comment by kpgarrod** [comment]
+1

**Comment by jayoungers** [comment]
+1.  Based on ng-conf in March I was figuring we'd be able to refactor from ui-router to this sometime in April

**Comment by thelgevold** [comment]
+1 I am definitely curious about the part of this question that talks about Angular 1.4 and the new router not being ready.....

**Comment by paulwalker** [comment]
+1, I have had to abandon use of it, mostly because of https://github.com/angular/router/issues/311. I'm using ui-router at this point, much better baked. I'm curious why, if it was considered at all, not adopted for angular 2.0. If the issue was an academic in it labeling itself as a "state" machine type routing, the fact that it's resolves and route configuration can now be defined in specific controller modules outweighs what I see so far in this new router as far as any best pattern practices I see being offered in this code base.

**Comment by KhoaSydney** [comment]
+1 
I have been tracking the progress for a while and I have to say it's slow :/

**Comment by rrajewski** [comment]
+1

**Comment by jamiecounsell** [comment]
+2 @shawn-simon-developer

**Comment by hannahhoward** [comment]
FYI I found this: https://www.youtube.com/watch?t=1956&v=sfV0ApyN5fM

That's about a week and a half old but it may provide some answers.

**Comment by twhitbeck** [comment]
thanks @hannahhoward that was indeed helpful.

Am I right in understand that active development on the router is going in angular/angular atm? But that at some point it's going to shift back to this repo?

**Comment by nikaspran** [comment]
That's the idea. However, looking at the [Angular weekly meeting notes](https://docs.google.com/document/d/150lerb1LmNLuau_a_EznPV1I1UHMTbEl61t4hZ7ZpS0/edit), it seems like the router won't be out for a while, ng1 or ng2:

> May 11th, 2015 (Proposed Agenda)
> * (igor): router
> Too many rough edges, not ready for v1.4. Work with Brian and Matias to improve it.
>  * Improve animation support. (directionality of the state transition)
>  * Improved migration story.

**Comment by grbush** [comment]
Ahh well, that's a shame. I'm thinking of checking out the angular / angular repository and seeing if I can build an es5 new router out of it. Perhaps it's not too broken for my current project, which is just switching between tabs and views in a single page app. I don't need really need animation or a migration story.

Has anyone else had any luck extracting useful es5 routing code from angular / angular?

**Comment by pumpk1n** [comment]
Haven't tried it with Angular 1.x, but for Angular 2 it's pretty much broken as well. AFAIK this is the latest version of the router https://github.com/btford/angular/tree/refactor-router-navigate, but I couldn't get it work with ng2.

Would be real nice to have a lil more info from @btford.

**Comment by twhitbeck** [comment]
The [Meeting Notes](https://docs.google.com/document/d/150lerb1LmNLuau_a_EznPV1I1UHMTbEl61t4hZ7ZpS0/edit) have a little more to say this week.

> * Brian and Matias will work with Igor on plug-ins for build improvements. File build issues under the "Build" milestone for angular 2 so we can find them.
> * Once typescriptification and build stability are in, router, decorators, custom render, and finalising the public api in the core (app injector, pipes, forms) are next in terms of urgency.

Looks like the router is taking a backseat to "typescriptification" atm...it sure is hard to keep up with the latest goings on.

**Comment by maku** [comment]
The whole router story is nothing but sad...

**Comment by johnkit** [comment]
+1

I switched to ngNewRouter for a new project, based on the ng-conf presentation too. So now I have to cut bait and retool with ui-router. Certainly discourages me from being an early-adopter in the future.

**Comment by khaledabbas** [comment]
+1

**Comment by dereklin** [comment]
+1

**Comment by frodefi** [comment]
@dereklin A few weeks ago the new router was a few weeks away, but now it is more uncertain. This new router would work both in Angular 1 and 2, so it depends on how urgent you need a router. I wanted to use it with my current project, but now I probably should change to some other router.

**Comment by grbush** [comment]
Yes, I was very hopeful about Component Router, but have just spent ~4hrs pulling it out and replacing it with ui.router. It's not too bad a transition to go back from Component Router to ui.router. I think starting with Component Router gave our app a good component-based structure and will be an easier transition to new router and angular2 once they're ready.

I'm guessing the angular guys have disappeared into a migrating-to-TypeScript hole (a.k.a typescriptification), wondering when they will emerge :)

Cheers,
Geoff

Geoff Bush / Development Manager / iLign Ltd
T: + 64 4 910 5688, M: +64 21 104 2878
E: geoff.bush@ilign.com W: www.ilign.com
Level 4, Anvil House, 138 – 140 Wakefield Street
PO Box 19-132, Te Aro, Wellington, 6149, New Zealand.
________________________________

**Comment by thelgevold** [comment]
I got think the proposed new router looks promising. Should be familiar to people who have used DurandalJS before.
Anyway, lazy loading does not seem to be slated for 1.4, but is instead targeted for 1.5
 
Got some more info regarding progress on the lazy loading component of the router via twitter

https://twitter.com/helgevold/status/600525678010839040

**Comment by dereklin** [comment]
Just switched from the new router to ui-router.  It was pretty quick.  Also, we had a problem with a value not binding in a directive using the new router.  It's fixed when using ui-router.  The new router is buggy.  Hope all the bugs will be fixed soon and new features will be added.  Then we'll have one router to rule them all... (That's the message I got when I first used the new router)

**Comment by dcunited08** [comment]
Router was pulled from release:
http://angularjs.blogspot.com/2015/05/angular-140-jaracimrman-existence.html

**Comment by frodefi** [comment]
Ok, so now the router has been officially pulled from the 1.4 release. But perhaps we at least could use the latest version of the new router, even if it is not 100% ready for official release? As I understand it, the latest version is not available here. The version that is here has too many limits for it to be used.

**Comment by frodefi** [comment]
I am hoping the activity on this router will not suffer from the similarity to t-fords...

**Comment by hannahhoward** [comment]
sorta update on remaining issues. Seems clear development is happening on angular/angular. https://github.com/angular/angular/issues/2329

**Comment by havenchyk** [comment]
Check the updates there https://github.com/angular/angular/issues/2329

**Comment by thelgevold** [comment]
The router is finally usable now with TypeScript typings. I have created an example here if you are interested:
http://www.syntaxsuccess.com/viewarticle/routing-in-angular-2.0

The code is also on Github (link in article)

**Comment by frodefi** [comment]
This is the [status](https://github.com/angular/angular/pull/3698) currently

**Comment by frodefi** [comment]
Now Angular issue [#3774](https://github.com/angular/angular/issues/3774) seems like the interesting one to keep an eye on, from an Angular 1 perspective...

**Comment by corydorning** [comment]
new code is here:
https://github.com/angular/angular/tree/master/modules/angular1_router

Anyone tried it with the examples and have success?
